tests/integration/ha_tests/helpers.py

```diff
@@ -279,9 +308,9 @@ async def mongod_ready(ops_test: OpsTest, unit: int) -> bool:
     return True
 
 
-async def get_replica_set_primary(ops_test: OpsTest) -> Optional[str]:
+async def get_replica_set_primary(ops_test: OpsTest, excluded: List[str] = []) -> Optional[str]:
```

Comment:
Passing an exclusion list, because when doing the network cut, we check for a new primary while the old one is still isolated.